packages/math-input/src/components/keypad/keypad-pages/operators-page.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from "react";
+
+import Keys from "../../../data/key-configs";
+import {ClickKeyCallback} from "../../../types";
+import {KeypadButton} from "../keypad-button";
+
+type Props = {
+    onClickKey: ClickKeyCallback;
+    preAlgebra?: boolean;
+    logarithms?: boolean;
+    basicRelations?: boolean;
+    advancedRelations?: boolean;
+};
+
+export default function OperatorsPage(props: Props) {
+    const {
+        onClickKey,
+        preAlgebra,
+        logarithms,
+        basicRelations,
+        advancedRelations,
+    } = props;
+
+    return (
+        <>
+            {/* Row 1 */}
+            {preAlgebra && (
+                <>
+                    <KeypadButton
+                        keyConfig={Keys.EXP_2}
+                        onClickKey={onClickKey}
+                        coord={[0, 0]}
+                    />
+                    <KeypadButton
+                        keyConfig={Keys.EXP}
+                        onClickKey={onClickKey}
+                        coord={[1, 0]}
+                    />
+                    <KeypadButton
+                        keyConfig={Keys.SQRT}
+                        onClickKey={onClickKey}
+                        coord={[2, 0]}
+                    />
+                    <KeypadButton
+                        keyConfig={Keys.RADICAL}
+                        onClickKey={onClickKey}
+                        coord={[3, 0]}
+                    />
+                </>
+            )}
+
+            {/* Row 2 */}
+            {logarithms && (
```

Comment:
If the parent turns off any of these options, are we ok with the rest of the buttons not moving up? We have hard-coded row numbers in these layouts so they won't automatically "move up" to fill in the space for button groups that are disabled.

Reply:
So that's how things were before this refactor: if we didn't render the buttons we'd render placeholders. This PR removes the need for the placeholders thanks to the coordinate positioning.

Personally I think this is best. As a learner, if I want the `log` button it's going to be in the same position no matter what else is showing. It's a small thing, but I think people build muscle memory around where things are and shifting that can be disorienting.

Reply:
I do agree.  Just wanted to double-check that's what we wanted. :)